core/src/main/scala/org/apache/spark/scheduler/LiveListenerBus.scala

```diff
@@ -186,6 +186,17 @@ private[spark] class LiveListenerBus(conf: SparkConf) {
     metricsSystem.registerSource(metrics)
   }
 
+  /**
+   * For testing only. Wait until there are no more events in the queue, or until the default
+   * wait time has elapsed. Throw `TimeoutException` if the specified time elapsed before the queue
+   * emptied.
+   * Exposed for testing.
+   */
+  @throws(classOf[TimeoutException])
+  def waitUntilEmpty(): Unit = {
+    waitUntilEmpty(TimeUnit.SECONDS.toMillis(10))
+  }
```

Comment:
Thank you for the follow-up. Could you file a JIRA issue since this seems too big for `MINOR`?

Reply:
OK, I'll file an issue and update the title of PR. Thanks!

Reply:
Thanks!